test/components/tree-view.spec.tsx

```diff
@@ -0,0 +1,109 @@
+
+    });
+
+    describe('Using default renderer', () => {
+
+        it('renders to the screen', () => {
+            const { select, waitForDom } = clientRenderer.render(<TreeView dataSource={treeData} />);
+
+            return waitForDom(() => expect(select(treeView)).to.be.present());
+        });
+
+        it('renders correct children', () => {
+            const { select, waitForDom } = clientRenderer.render(<TreeView dataSource={treeData} />);
+
+            return waitForDom(() =>
+                treeData.forEach((item: TreeItemData) =>
+                    expect(select(treeView, getTreeItem(item.label))).to.be.present()));
+        });
+
+        it('invokes the onSelectItem callback when an item is clicked', () => {
+            const onSelectItem = sinon.spy();
+            const { select } = clientRenderer.render(<TreeView dataSource={treeData} onSelectItem={onSelectItem}/>);
+
+            const firstNodeChildren = treeData[0].children;
+
+            const nodeLabelToClick = getTreeItem(firstNodeChildren![1].label);
+
+            simulate.click(select(nodeLabelToClick));
+
+            return waitFor(() => {
+                expect(onSelectItem).to.have.been.calledOnce;
+                expect(onSelectItem).to.have.been.calledWithMatch(firstNodeChildren![1]);
+            });
+        });
+
+        describe('<TreeItem />', () => {
+
+            const item = { label: 'label' };
+
+            it('renders an item', () => {
+                const { select, waitForDom } =
+                    clientRenderer.render(<TreeItem item={item} itemRenderer={TreeItem} isSelected={noop}/>);
+
+                return waitForDom(() => expect(select(getTreeItem(item.label))).to.be.present());
+            });
+
+            it('renders with provided label', () => {
+                const { select, waitForDom } =
+                    clientRenderer.render(<TreeItem item={item} itemRenderer={TreeItem} isSelected={noop}/>);
+
+                return waitForDom(() => expect(select(getTreeItem(item.label) + '_LABEL')).to.have.text(item.label));
+            });
+
+            it('renders with an icon', () => {
+                const { select, waitForDom } =
+                    clientRenderer.render(<TreeItem item={item} itemRenderer={TreeItem} isSelected={noop}/>);
+
+                return waitForDom(() => expect(select(getTreeItem(item.label) + '_ICON')).to.be.present());
+            });
+
+            it('invokes onClick when clicked', () => {
+                const onClick = sinon.spy();
+                const { select } = clientRenderer.render(<TreeItem item={item} itemRenderer={TreeItem}
+                                                                   onItemClick={onClick} isSelected={noop}/>);
+
+                simulate.click(select(getTreeItem(item.label)));
+
+                return waitFor(() => expect(onClick).to.have.been.calledOnce);
```

Comment:
i think i have it covered in 'renders correct children' for the TreeView / Using default renderer, but just in case someone decides to use the tree item by itself, adding the test for the children